gobblin-runtime/src/main/java/org/apache/gobblin/runtime/job_catalog/FSJobCatalog.java

```diff
@@ -108,8 +115,10 @@ public synchronized void put(JobSpec jobSpec) {
     Preconditions.checkState(state() == State.RUNNING, String.format("%s is not running.", this.getClass().getName()));
     Preconditions.checkNotNull(jobSpec);
     try {
+      long startTime = System.currentTimeMillis();
       Path jobSpecPath = getPathForURI(this.jobConfDirPath, jobSpec.getUri());
       materializedJobSpec(jobSpecPath, jobSpec, this.fs);
+      ((MutableStandardMetrics)this.metrics).updatePutJobTime(startTime);
```

Comment:
How about declaring a final field to store the `MutableStandardMetrics` object? In this case, you don't need to cast `this.metrics` every time and worry about potential CAST exception.